### PR TITLE
Async

### PR DESCRIPTION
**Pull request by grinderrz**
Support for ring async handlers.

### CODE REVIEW

**Comment by nberger** [comment]
Thanks for this! This was a missing piece for a long time.

From first sight it looks good, I'll review and try it soon.

File: src/ring/logger.clj [hunk 1]
```diff
@@ -6,37 +6,66 @@
...
+                  (messages/finished options request %)
+                  (respond %))
+               #(let [request (assoc-end-time request timing)]
+                  (messages/exception options request %)
+                  (raise %)))
+      (catch Throwable t
+        (when exceptions
+          (let [request (assoc-end-time request timing)]
+            (messages/exception options request t)))
+        (throw t))))))
```
Comment (nberger) [inline src/ring/logger.clj]: I think we should call `raise` here instead of `throw`ing

File: src/ring/logger.clj [hunk 2]
```diff
@@ -6,37 +6,66 @@
...
+      (messages/request-details options request)
+      (messages/request-params options request)
+
+      (handler request
+               #(let [request (assoc-end-time request timing)]
+                  (messages/sending-response options %)
+                  (messages/finished options request %)
+                  (respond %))
+               #(let [request (assoc-end-time request timing)]
+                  (messages/exception options request %)
```
Comment (nberger) [inline src/ring/logger.clj]: We want to log the exception by calling `messages/exception` only when `exceptions` is truthy (it's an option). It should be similar I think to what's being done in the `catch` few lines below

**Comment by nberger** [comment]
I just added two comments related to exception handling. If you could fix those, I'll merge it. Thanks again!

**Comment by grinderrz** [comment]
It's fixed now, if I understood your comments correctly.

**Comment by nberger** [comment]
Looks good, thanks!

**Pull request merged**